### PR TITLE
EZP-29713: As an Editor, I want to Move multiple content items at a time

### PR DESCRIPTION
**Pull request by ViniTou**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-29713
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

## Description

Proposed solution creates API endpoint that in a payload can accept all structured requests made to any kernel api endpoint. `content` parameter must be properly escaped xml or json string. Keys name are not connected anyhow to type of operations i.e. you can name them however you want as long as there are unique.
 
## Example

```
POST http://localhost:8080/admin/api/bulk
Cache-Control: no-cache
Content-Type: application/vnd.ez.api.BulkOperation+json
Accept: application/vnd.ez.api.BulkOperationResponse+json
```

Payload:
```json
{
  "bulkOperations": {
    "operations": {
        "deleteOperation" : {
            "method": "DELETE",
            "uri": "api/ezp/v2/content/trash"
        },
        "getOperation" : {
            "uri":"api/ezp/v2/content/objects/1539",
            "method":"GET",
            "headers": {
               "ACCEPT":"application/vnd.ez.api.Content+json"
            }
        },
        "createContentTypeOperation" : {
            "content": "{\"ContentTypeGroupInput\": { \"identifier\": \"neverGonnaGiveYouUp\" }}",
            "headers": {
                "CONTENT-TYPE": "application/vnd.ez.api.ContentTypeGroupInput+json",
                "ACCEPT": "application/vnd.ez.api.ContentTypeGroup+json"
            },
            "method": "POST",
            "uri": "api/ezp/v2/content/typegroups"
        }
    }
  }
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bulkOperations>
   <operations>
      <createContentTypeOperation>
         <content>&lt;ContentTypeGroupInput&gt;                 &lt;identifier&gt;neverGonnaMakeYouCry&lt;/identifier&gt;              &lt;/ContentTypeGroupInput&gt;</content>
         <headers>
            <ACCEPT>application/vnd.ez.api.ContentTypeGroup+xml</ACCEPT>
            <CONTENT-TYPE>application/vnd.ez.api.ContentTypeGroupInput+xml</CONTENT-TYPE>
         </headers>
         <method>POST</method>
         <uri>api/ezp/v2/content/typegroups</uri>
      </createContentTypeOperation>
      <deleteOperation>
         <method>DELETE</method>
         <uri>api/ezp/v2/content/trash</uri>
      </deleteOperation>
      <getOperation>
         <headers>
            <ACCEPT>application/vnd.ez.api.Content+xml</ACCEPT>
         </headers>
         <method>GET</method>
         <uri>api/ezp/v2/content/objects/52</uri>
      </getOperation>
   </operations>
</bulkOperations>
```

Response:
```json

{
    "BulkOperationResponse": {
        "_media-type": "application\/vnd.ez.api.BulkOperationResponse+json",
        "operations": {
            "deleteOperation": {
                "_media-type": "application\/vnd.ez.api.OperationResponse+json",
                "statusCode": 204,
                "headers": {
                    "cache-control": "no-cache, private",
                    "date": "Thu, 25 Oct 2018 08:37:29 GMT",
                    "x-cache-debug": 1,
                    "x-debug-token": "9efde7",
                    "x-debug-token-link": "http:\/\/localhost\/_profiler\/9efde7"
                },
                "content": ""
            },
            "getOperation": {
                "_media-type": "application\/vnd.ez.api.OperationResponse+json",
                "statusCode": 200,
                "headers": {
                    "cache-control": "public, s-maxage=86400",
                    "date": "Thu, 25 Oct 2018 08:37:29 GMT",
                    "content-type": "application\/vnd.ez.api.Content+json",
                    "accept-patch": "application\/vnd.ez.api.ContentUpdate+json",
                    "vary": "Accept",
                    "x-cache-debug": 1,
                    "x-debug-token": "2c54a1",
                    "x-debug-token-link": "http:\/\/localhost\/_profiler\/2c54a1"
                },
                "content": "{\n    \"Content\": {\n        \"_media-type\": \"application\\\/vnd.ez.api.Content+json\",\n        \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\",\n        \"_remoteId\": \"34720ff636e1d4ce512f762dc638e4ac\",\n        \"_id\": 52,\n        \"ContentType\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.ContentType+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/types\\\/42\"\n        },\n        \"Name\": \"Home\",\n        \"Versions\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.VersionList+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/versions\"\n        },\n        \"CurrentVersion\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.Version+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/currentversion\",\n            \"Version\": {\n                \"_media-type\": \"application\\\/vnd.ez.api.Version+json\",\n                \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/versions\\\/1\",\n                \"VersionInfo\": {\n                    \"id\": 512,\n                    \"versionNo\": 1,\n                    \"status\": \"PUBLISHED\",\n                    \"modificationDate\": \"2015-09-17T09:22:23+00:00\",\n                    \"Creator\": {\n                        \"_media-type\": \"application\\\/vnd.ez.api.User+json\",\n                        \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/user\\\/users\\\/14\"\n                    },\n                    \"creationDate\": \"2015-09-17T09:22:22+00:00\",\n                    \"initialLanguageCode\": \"eng-GB\",\n                    \"languageCodes\": \"eng-GB\",\n                    \"VersionTranslationInfo\": {\n                        \"_media-type\": \"application\\\/vnd.ez.api.VersionTranslationInfo+json\",\n                        \"Language\": [\n                            {\n                                \"languageCode\": \"eng-GB\"\n                            }\n                        ]\n                    },\n                    \"names\": {\n                        \"value\": [\n                            {\n                                \"_languageCode\": \"eng-GB\",\n                                \"#text\": \"Home\"\n                            }\n                        ]\n                    },\n                    \"Content\": {\n                        \"_media-type\": \"application\\\/vnd.ez.api.ContentInfo+json\",\n                        \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\"\n                    }\n                },\n                \"Fields\": {\n                    \"field\": [\n                        {\n                            \"id\": 242,\n                            \"fieldDefinitionIdentifier\": \"name\",\n                            \"languageCode\": \"eng-GB\",\n                            \"fieldTypeIdentifier\": \"ezstring\",\n                            \"fieldValue\": \"Home\"\n                        },\n                        {\n                            \"id\": 243,\n                            \"fieldDefinitionIdentifier\": \"description\",\n                            \"languageCode\": \"eng-GB\",\n                            \"fieldTypeIdentifier\": \"ezstring\",\n                            \"fieldValue\": \"Home Page\"\n                        },\n                        {\n                            \"id\": 244,\n                            \"fieldDefinitionIdentifier\": \"page\",\n                            \"languageCode\": \"eng-GB\",\n                            \"fieldTypeIdentifier\": \"ezlandingpage\",\n                            \"fieldValue\": {\n                                \"layout\": \"default\",\n                                \"zones\": [\n                                    {\n                                        \"id\": \"1\",\n                                        \"name\": \"default\",\n                                        \"blocks\": [\n                                            {\n                                                \"visible\": true,\n                                                \"id\": \"1\",\n                                                \"type\": \"tag\",\n                                                \"name\": \"Tag\",\n                                                \"view\": \"default\",\n                                                \"class\": null,\n                                                \"style\": null,\n                                                \"compiled\": null,\n                                                \"since\": null,\n                                                \"till\": null,\n                                                \"attributes\": [\n                                                    {\n                                                        \"id\": \"1\",\n                                                        \"name\": \"content\",\n                                                        \"value\": \"<h1>eZ Studio<\\\/h1> <p>This is the clean install coming with eZ Studio. Now you can start creating your own design.<\\\/p>\"\n                                                    }\n                                                ]\n                                            }\n                                        ]\n                                    }\n                                ]\n                            }\n                        }\n                    ]\n                },\n                \"Relations\": {\n                    \"_media-type\": \"application\\\/vnd.ez.api.RelationList+json\",\n                    \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/versions\\\/1\\\/relations\",\n                    \"Relation\": []\n                }\n            }\n        },\n        \"Section\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.Section+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/sections\\\/1\"\n        },\n        \"MainLocation\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.Location+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/locations\\\/1\\\/2\"\n        },\n        \"Locations\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.LocationList+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/locations\"\n        },\n        \"Owner\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.User+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/user\\\/users\\\/14\"\n        },\n        \"lastModificationDate\": \"2015-09-17T09:22:23+00:00\",\n        \"publishedDate\": \"2015-09-17T09:22:23+00:00\",\n        \"mainLanguageCode\": \"eng-GB\",\n        \"currentVersionNo\": 1,\n        \"alwaysAvailable\": true,\n        \"ObjectStates\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.ContentObjectStates+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/objects\\\/52\\\/objectstates\"\n        }\n    }\n}"
            },
            "createContentTypeOperation": {
                "_media-type": "application\/vnd.ez.api.OperationResponse+json",
                "statusCode": 201,
                "headers": {
                    "cache-control": "no-cache, private",
                    "date": "Thu, 25 Oct 2018 08:37:29 GMT",
                    "content-type": "application\/vnd.ez.api.ContentTypeGroup+json",
                    "accept-patch": "application\/vnd.ez.api.ContentTypeGroupInput+json",
                    "location": "\/api\/ezp\/v2\/content\/typegroups\/8",
                    "x-cache-debug": 1,
                    "x-debug-token": "5296e6",
                    "x-debug-token-link": "http:\/\/localhost\/_profiler\/5296e6",
                    "set-cookie": "sf_redirect=%7B%22token%22%3A%225296e6%22%2C%22route%22%3A%22ezpublish_rest_createContentTypeGroup%22%2C%22method%22%3A%22POST%22%2C%22controller%22%3A%22ezpublish_rest.controller.content_type%3AcreateContentTypeGroup%22%2C%22status_code%22%3A201%2C%22status_text%22%3A%22Created%22%7D; path=\/; httponly"
                },
                "content": "{\n    \"ContentTypeGroup\": {\n        \"_media-type\": \"application\\\/vnd.ez.api.ContentTypeGroup+json\",\n        \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/typegroups\\\/8\",\n        \"id\": \"8\",\n        \"identifier\": \"neverGonnaGiveYouUp\",\n        \"created\": \"2018-10-25T08:37:29+00:00\",\n        \"modified\": \"2018-10-25T08:37:29+00:00\",\n        \"Creator\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.User+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/user\\\/users\\\/14\"\n        },\n        \"Modifier\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.User+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/user\\\/users\\\/14\"\n        },\n        \"ContentTypes\": {\n            \"_media-type\": \"application\\\/vnd.ez.api.ContentTypeInfoList+json\",\n            \"_href\": \"\\\/api\\\/ezp\\\/v2\\\/content\\\/typegroups\\\/8\\\/types\"\n        }\n    }\n}"
            }
        }
    }
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<BulkOperationResponse media-type="application/vnd.ez.api.BulkOperationResponse+xml">
 <operations>
  <createContentTypeOperation media-type="application/vnd.ez.api.OperationResponse+xml">
   <statusCode>201</statusCode>
   <headers>
    <cache-control>no-cache, private</cache-control>
    <date>Thu, 25 Oct 2018 08:33:20 GMT</date>
    <content-type>application/vnd.ez.api.ContentTypeGroup+xml</content-type>
    <accept-patch>application/vnd.ez.api.ContentTypeGroupInput+xml</accept-patch>
    <location>/api/ezp/v2/content/typegroups/6</location>
    <x-cache-debug>1</x-cache-debug>
    <x-debug-token>94eeab</x-debug-token>
    <x-debug-token-link>http://localhost/_profiler/94eeab</x-debug-token-link>
    <set-cookie>sf_redirect=%7B%22token%22%3A%2294eeab%22%2C%22route%22%3A%22ezpublish_rest_createContentTypeGroup%22%2C%22method%22%3A%22POST%22%2C%22controller%22%3A%22ezpublish_rest.controller.content_type%3AcreateContentTypeGroup%22%2C%22status_code%22%3A201%2C%22status_text%22%3A%22Created%22%7D; path=/; httponly</set-cookie>
   </headers>
   <content>&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;ContentTypeGroup media-type=&quot;application/vnd.ez.api.ContentTypeGroup+xml&quot; href=&quot;/api/ezp/v2/content/typegroups/6&quot;&gt;
 &lt;id&gt;6&lt;/id&gt;
 &lt;identifier&gt;neverGonnaMakeYouCry&lt;/identifier&gt;
 &lt;created&gt;2018-10-25T08:33:20+00:00&lt;/created&gt;
 &lt;modified&gt;2018-10-25T08:33:20+00:00&lt;/modified&gt;
 &lt;Creator media-type=&quot;application/vnd.ez.api.User+xml&quot; href=&quot;/api/ezp/v2/user/users/14&quot;/&gt;
 &lt;Modifier media-type=&quot;application/vnd.ez.api.User+xml&quot; href=&quot;/api/ezp/v2/user/users/14&quot;/&gt;
 &lt;ContentTypes media-type=&quot;application/vnd.ez.api.ContentTypeInfoList+xml&quot; href=&quot;/api/ezp/v2/content/typegroups/6/types&quot;/&gt;
&lt;/ContentTypeGroup&gt;
</content>
  </createContentTypeOperation>
  <deleteOperation media-type="application/vnd.ez.api.OperationResponse+xml">
   <statusCode>204</statusCode>
   <headers>
    <cache-control>no-cache, private</cache-control>
    <date>Thu, 25 Oct 2018 08:33:20 GMT</date>
    <x-cache-debug>1</x-cache-debug>
    <x-debug-token>5a4a54</x-debug-token>
    <x-debug-token-link>http://localhost/_profiler/5a4a54</x-debug-token-link>
   </headers>
   <content></content>
  </deleteOperation>
  <getOperation media-type="application/vnd.ez.api.OperationResponse+xml">
   <statusCode>200</statusCode>
   <headers>
    <cache-control>public, s-maxage=86400</cache-control>
    <date>Thu, 25 Oct 2018 08:33:20 GMT</date>
    <content-type>application/vnd.ez.api.Content+xml</content-type>
    <accept-patch>application/vnd.ez.api.ContentUpdate+xml</accept-patch>
    <vary>Accept</vary>
    <x-cache-debug>1</x-cache-debug>
    <x-debug-token>7d9070</x-debug-token>
    <x-debug-token-link>http://localhost/_profiler/7d9070</x-debug-token-link>
   </headers>
   <content>&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;
&lt;Content media-type=&quot;application/vnd.ez.api.Content+xml&quot; href=&quot;/api/ezp/v2/content/objects/52&quot; remoteId=&quot;34720ff636e1d4ce512f762dc638e4ac&quot; id=&quot;52&quot;&gt;
 &lt;ContentType media-type=&quot;application/vnd.ez.api.ContentType+xml&quot; href=&quot;/api/ezp/v2/content/types/42&quot;/&gt;
 &lt;Name&gt;Home&lt;/Name&gt;
 &lt;Versions media-type=&quot;application/vnd.ez.api.VersionList+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/versions&quot;/&gt;
 &lt;CurrentVersion media-type=&quot;application/vnd.ez.api.Version+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/currentversion&quot;&gt;
  &lt;Version media-type=&quot;application/vnd.ez.api.Version+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/versions/1&quot;&gt;
   &lt;VersionInfo&gt;
    &lt;id&gt;512&lt;/id&gt;
    &lt;versionNo&gt;1&lt;/versionNo&gt;
    &lt;status&gt;PUBLISHED&lt;/status&gt;
    &lt;modificationDate&gt;2015-09-17T09:22:23+00:00&lt;/modificationDate&gt;
    &lt;Creator media-type=&quot;application/vnd.ez.api.User+xml&quot; href=&quot;/api/ezp/v2/user/users/14&quot;/&gt;
    &lt;creationDate&gt;2015-09-17T09:22:22+00:00&lt;/creationDate&gt;
    &lt;initialLanguageCode&gt;eng-GB&lt;/initialLanguageCode&gt;
    &lt;languageCodes&gt;eng-GB&lt;/languageCodes&gt;
    &lt;VersionTranslationInfo media-type=&quot;application/vnd.ez.api.VersionTranslationInfo+xml&quot;&gt;
     &lt;Language&gt;
      &lt;languageCode&gt;eng-GB&lt;/languageCode&gt;
     &lt;/Language&gt;
    &lt;/VersionTranslationInfo&gt;
    &lt;names&gt;
     &lt;value languageCode=&quot;eng-GB&quot;&gt;Home&lt;/value&gt;
    &lt;/names&gt;
    &lt;Content media-type=&quot;application/vnd.ez.api.ContentInfo+xml&quot; href=&quot;/api/ezp/v2/content/objects/52&quot;/&gt;
   &lt;/VersionInfo&gt;
   &lt;Fields&gt;
    &lt;field&gt;
     &lt;id&gt;242&lt;/id&gt;
     &lt;fieldDefinitionIdentifier&gt;name&lt;/fieldDefinitionIdentifier&gt;
     &lt;languageCode&gt;eng-GB&lt;/languageCode&gt;
     &lt;fieldTypeIdentifier&gt;ezstring&lt;/fieldTypeIdentifier&gt;
     &lt;fieldValue&gt;Home&lt;/fieldValue&gt;
    &lt;/field&gt;
    &lt;field&gt;
     &lt;id&gt;243&lt;/id&gt;
     &lt;fieldDefinitionIdentifier&gt;description&lt;/fieldDefinitionIdentifier&gt;
     &lt;languageCode&gt;eng-GB&lt;/languageCode&gt;
     &lt;fieldTypeIdentifier&gt;ezstring&lt;/fieldTypeIdentifier&gt;
     &lt;fieldValue&gt;Home Page&lt;/fieldValue&gt;
    &lt;/field&gt;
    &lt;field&gt;
     &lt;id&gt;244&lt;/id&gt;
     &lt;fieldDefinitionIdentifier&gt;page&lt;/fieldDefinitionIdentifier&gt;
     &lt;languageCode&gt;eng-GB&lt;/languageCode&gt;
     &lt;fieldTypeIdentifier&gt;ezlandingpage&lt;/fieldTypeIdentifier&gt;
     &lt;fieldValue&gt;
      &lt;value key=&quot;layout&quot;&gt;default&lt;/value&gt;
      &lt;value key=&quot;zones&quot;&gt;
       &lt;value&gt;
        &lt;value key=&quot;id&quot;&gt;1&lt;/value&gt;
        &lt;value key=&quot;name&quot;&gt;default&lt;/value&gt;
        &lt;value key=&quot;blocks&quot;&gt;
         &lt;value&gt;
          &lt;value key=&quot;visible&quot;&gt;true&lt;/value&gt;
          &lt;value key=&quot;id&quot;&gt;1&lt;/value&gt;
          &lt;value key=&quot;type&quot;&gt;tag&lt;/value&gt;
          &lt;value key=&quot;name&quot;&gt;Tag&lt;/value&gt;
          &lt;value key=&quot;view&quot;&gt;default&lt;/value&gt;
          &lt;value key=&quot;class&quot;/&gt;
          &lt;value key=&quot;style&quot;/&gt;
          &lt;value key=&quot;compiled&quot;/&gt;
          &lt;value key=&quot;since&quot;/&gt;
          &lt;value key=&quot;till&quot;/&gt;
          &lt;value key=&quot;attributes&quot;&gt;
           &lt;value&gt;
            &lt;value key=&quot;id&quot;&gt;1&lt;/value&gt;
            &lt;value key=&quot;name&quot;&gt;content&lt;/value&gt;
            &lt;value key=&quot;value&quot;&gt;&amp;lt;h1&amp;gt;eZ Studio&amp;lt;/h1&amp;gt; &amp;lt;p&amp;gt;This is the clean install coming with eZ Studio. Now you can start creating your own design.&amp;lt;/p&amp;gt;&lt;/value&gt;
           &lt;/value&gt;
          &lt;/value&gt;
         &lt;/value&gt;
        &lt;/value&gt;
       &lt;/value&gt;
      &lt;/value&gt;
     &lt;/fieldValue&gt;
    &lt;/field&gt;
   &lt;/Fields&gt;
   &lt;Relations media-type=&quot;application/vnd.ez.api.RelationList+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/versions/1/relations&quot;/&gt;
  &lt;/Version&gt;
 &lt;/CurrentVersion&gt;
 &lt;Section media-type=&quot;application/vnd.ez.api.Section+xml&quot; href=&quot;/api/ezp/v2/content/sections/1&quot;/&gt;
 &lt;MainLocation media-type=&quot;application/vnd.ez.api.Location+xml&quot; href=&quot;/api/ezp/v2/content/locations/1/2&quot;/&gt;
 &lt;Locations media-type=&quot;application/vnd.ez.api.LocationList+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/locations&quot;/&gt;
 &lt;Owner media-type=&quot;application/vnd.ez.api.User+xml&quot; href=&quot;/api/ezp/v2/user/users/14&quot;/&gt;
 &lt;lastModificationDate&gt;2015-09-17T09:22:23+00:00&lt;/lastModificationDate&gt;
 &lt;publishedDate&gt;2015-09-17T09:22:23+00:00&lt;/publishedDate&gt;
 &lt;mainLanguageCode&gt;eng-GB&lt;/mainLanguageCode&gt;
 &lt;currentVersionNo&gt;1&lt;/currentVersionNo&gt;
 &lt;alwaysAvailable&gt;true&lt;/alwaysAvailable&gt;
 &lt;ObjectStates media-type=&quot;application/vnd.ez.api.ContentObjectStates+xml&quot; href=&quot;/api/ezp/v2/content/objects/52/objectstates&quot;/&gt;
&lt;/Content&gt;
</content>
  </getOperation>
 </operations>
</BulkOperationResponse>
```

## TODO:
- [x] Add xml handling here:
https://github.com/ViniTou/ezplatform-admin-ui/commit/aeb283ef812727adfe4a901bae79c693c3d5ef1c#diff-1376eab2b17204841171cef676fc9845R61

- [x] Add consistent handling for HTTP headers inside payload.

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by ViniTou** [comment]
@adamwojs 
@Nattfarinn 
@mikadamczyk 

and @tischsoic cuz I changed request structure to accommodate requestId.

**Comment by ViniTou** [comment]
I just squashed commits into one

@micszo

File: src/lib/EventListener/ApiRequestListener.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace EzSystems\EzPlatformAdminUi\EventListener;
+
+use eZ\Bundle\EzPublishRestBundle\EventListener\RequestListener;
+use Symfony\Component\HttpFoundation\Request;
+
+class ApiRequestListener extends RequestListener
+{
+    // siteaccesname/api/method_name
+    const ADMIN_UI_REST_PREFIX_PATTERN = '/[a-zA-Z0-9-_]+\/api\/[a-zA-Z0-9-_]+/';
```
Comment (emodric) [inline src/lib/EventListener/ApiRequestListener.php]: Hi. This regex pattern is too broad. Upgrading eZ Platform to 7.4 RC broke a wide range of our REST routes in the following formats:

* 'https://example.com/netgen/api/some/path'
* 'https://example.com/admin/netgen/api/some/path'
* 'https://example.com/cro/netgen/api/some/path'

Comment (emodric) [reply src/lib/EventListener/ApiRequestListener.php]: Reported as https://jira.ez.no/browse/EZP-29933

**Pull request merged**